pycqed/analysis_v2/timedomain_analysis.py

```diff
@@ -289,11 +289,66 @@ def extract_data(self):
         if len(self.channel_map) == 0:
             raise ValueError('No qubit RO channels have been found.')
 
-        self.data_to_fit = deepcopy(self.get_param_value('data_to_fit', {}))
+        # whether to rotate data or not
+        self.rotate = self.get_param_value('rotate', default_value=True)
+
+        # to be used if data_type == singleshot (see process_data)
+        self.predict_proba = self.get_param_value("predict_proba", False)
+        if self.predict_proba and self.get_param_value("classified_ro", False):
+            log.warning("predict_proba set to 'False' as probabilities are"
+                        "already obtained from classified readout")
+            self.predict_proba = False
+        # ensure rotation is removed when single shots yield probabilities
+        if self.get_param_value("classified_ro", False) or self.predict_proba:
+            self.rotate = False
+
+        if self.get_param_value('global_PCA') is not None:
+            log.warning('Parameter "global_PCA" is deprecated. Please set '
+                        'rotation_type="global_PCA" instead.')
+        self.rotation_type = self.get_param_value(
+            'rotation_type',
+            default_value='cal_states' if self.rotate else 'no_rotation')
 
         # creates self.sp
         self.get_sweep_points()
 
+        # creates self.cp if found else defaults to legacy extraction
+        self.get_cal_points()
+
+        # get data_to_fit
+        self.data_to_fit = deepcopy(self.get_param_value('data_to_fit'))
```

Comment:
Is such a deepcopy also needed for other parameters, e.g. for cal points (see below)?

Reply:
I don't think this is needed here either. I will test